shared/common-adapters/markdown.shared.js

```diff
@@ -30,9 +30,16 @@ function processAST(ast, createComponent) {
   return ast.component
 }
 
+// Quick check to avoid markdown parsing overhead
+const maybeMarkdown = markdown => {
+  return markdown && !markdown.match(/^[^`\\~*_:>]*$/)
```

Comment:
This will not match:

 * `test.com`
 * `www.test.com/foo`
 * `😎` (unicode emoji)